dotcom-rendering/src/server/lib/logging-middleware.ts

```diff
@@ -12,38 +18,56 @@ const hasPageId = (body: unknown): body is { pageId: string } => {
 };
 
 /**
- * An Express middleware which handles creating our logger store and logging requests after they've
- * completed.
+ * An Express middleware which wraps the inbuilt json() body-parser middleware to record execution time.
  */
-export const requestLoggerMiddleware: RequestHandler = (req, res, next) => {
+const expressJsonMiddleware = express.json();
+export const timedExpressJsonMiddleware: RequestHandler = (req, res, next) => {
+	recordTimeStart('json');
+
+	// Pass a wrapper around Express's next() callback which record when expressJsonMiddleware calls next()
+	expressJsonMiddleware(req, res, () => {
+		recordTimeStop('json');
+		next();
+	});
+};
+
+/**
+ * An Express middleware which handles creating a logger store to save information to be logged in the request logger.
+ */
+export const loggingStoreMiddleware: RequestHandler = (req, res, next) => {
 	const loggerState = {
 		request: {
-			pageId: hasPageId(req.body) ? req.body.pageId : 'no-page-id-found',
+			pageId: 'before-request-logger-middleware',
 			path: req.path,
 			method: req.method,
 		},
-		timing: {},
+		timing: {
+			total: Date.now(),
```

Comment:
Supported in Node, `performance.now()` is more granular than `Date.now()`.

```suggestion
			total: performance.now(),
```